### PR TITLE
 On branch x86_64

### PR DESCRIPTION
**Pull request by monkeyx-net**
modified:   PortMaster/device_info.txt
	new file:   PortMaster/gptokeyb2.x86_64
	modified:   PortMaster/pylibs/harbourmaster/config.py
	modified:   PortMaster/pylibs/harbourmaster/platform.py
	new file:   PortMaster/retrodeck/PortMaster.txt
	new file:   PortMaster/retrodeck/control.txt

I now have Portmaster running via RetroDeck, PortMaster.sh and pugwash without errors.

I need to have another look and experiment with the do_realease for x86_64.

So will test that next.

I can imagine that retrodeck/PortMaster.txt and retrodeck/control.txt Might need some more stuff removing or adding.

I think this is functional enough to test ports running so will look at that know too.

PS do I need to add custom gamelist code as muos too?

### CODE REVIEW

**Comment by monkeyx-net** [comment]
PPS I am manually copying retrodeck PortMaster.sh and control.txt files for now too.

**Review by kloptops — CHANGES_REQUESTED** [review]
You need to copy the `install_portmaster` from the `PlatformmuOS` and add it to the `PlatformRetroDECK`, then modify it there. Otherwise it looks good. 👍

File: PortMaster/pylibs/harbourmaster/platform.py [hunk 1]
```diff
@@ -426,9 +433,15 @@ def portmaster_install(self):
         Move files into place.
         """
 
+        RD_DIR = self.hm.tools_dir / "PortMaster" / "retrodeck"
```
[the remark this answers is not in the document]
Comment (monkeyx-net) [reply PortMaster/pylibs/harbourmaster/platform.py:436]: > Why is this code in the PlatformmuOS, not in the PlatformRetroDECK? :)

Just looking a this now. That's why we need curly braces in all languages.

I missed a tab or a space :)

File: PortMaster/pugwash [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env python3
 
 ## -- BEGIN PORTMASTER INFO --
-PORTMASTER_VERSION = '2023-01-07-1007'
-PORTMASTER_RELEASE_CHANNEL = 'developer'
+PORTMASTER_VERSION = '2024.03.21-1440'
```
Comment (kloptops) [inline PortMaster/pugwash:4]: This needs reverting.

File: PortMaster/pylibs/harbourmaster/platform.py [hunk 3]
```diff
@@ -340,6 +340,56 @@ class PlatformRetroDECK(PlatformBase):
     def gamelist_file(self):
         return self.hm.ports_dir / 'gamelist.xml'
 
+    def gamelist_add(self, gameinfo_file):
```
Comment (kloptops) [inline PortMaster/pylibs/harbourmaster/platform.py:343]: this code is specifically for muOS so it can be omitted. :)

**Review by kloptops — CHANGES_REQUESTED** [review]
Only two changes this time! :)

**Pull request merged**